src/net.cpp

```diff
@@ -427,8 +427,16 @@ void CNode::PushVersion()
         LogPrint("net", "send version message: version %d, blocks=%d, us=%s, them=%s, peer=%d\n", PROTOCOL_VERSION, nBestHeight, addrMe.ToString(), addrYou.ToString(), id);
     else
         LogPrint("net", "send version message: version %d, blocks=%d, us=%s, peer=%d\n", PROTOCOL_VERSION, nBestHeight, addrMe.ToString(), id);
-    PushMessage("version", PROTOCOL_VERSION, nLocalServices, nTime, addrYou, addrMe,
+
+    // Stealth mode: pretend to be like Bitcoin Core to hide from DoS attackers.
+    if (IsStealthMode()) {
+        uint64_t services = NODE_NETWORK;
+        PushMessage("version", 70002, services, nTime, addrYou, addrMe,
+                nLocalHostNonce, FormatSubVersion("Satoshi", CLIENT_VERSION, std::vector<string>(), ""), nBestHeight, true);        
```

Comment:
These may warrant their own macros for cleanness, like: `STEALTH_PROTOCOL_VERSION` `STEALTH_CLIENT_NAME`. Will be better for find/replace too when the time comes to remove the feature.